crates/collab/src/db/queries/projects.rs

```diff
@@ -476,6 +477,60 @@ impl Database {
         .await
     }
 
+    pub async fn update_breakpoints(
+        &self,
+        connection_id: ConnectionId,
+        update: &proto::SynchronizeBreakpoints,
+    ) -> Result<TransactionGuard<HashSet<ConnectionId>>> {
+        let project_id = ProjectId::from_proto(update.project_id);
+        self.project_transaction(project_id, |tx| async move {
+            let project_path = update
+                .project_path
+                .as_ref()
+                .ok_or_else(|| anyhow!("invalid project path"))?;
+
+            // Ensure the update comes from the host.
+            let project = project::Entity::find_by_id(project_id)
+                .one(&*tx)
+                .await?
+                .ok_or_else(|| anyhow!("no such project"))?;
+
+            // remove all existing breakpoints
+            breakpoints::Entity::delete_many()
```

Comment:
I think we should be deleting all breakpoints associated with the project_id and project_path instead of just the project_id. 

If I'm understanding this correctly, update only contains a list of breakpoints per one file. So when the breakpoints from update are inserted into the db, breakpoints from other file's won't be updated correctly and would be deleted.

Reply:
Oops, yeah correct!